File: scala/bob/src/main/scala/Bob.scala
object Bob {


  def response(statement: String): String = statement.replaceAll("\\s+", "") match {
    case x if x.isEmpty => "Fine. Be that way!"
    case x if (x.toUpperCase equals x) && (x endsWith "?") && !(x.toLowerCase equals x) => "Calm down, I know what I'm doing!"
    case x if (x.toUpperCase equals x) && !(x.toLowerCase equals x) => "Whoa, chill out!"
    case x if x endsWith "?" => "Sure."
    case _ => "Whatever."
  }

}

File: csharp/bob/Bob.cs
using System;

public static class Bob
{
    public static string Response(string statement)
    {
        String sTrim = statement.Trim();
        if (sTrim.Length == 0)
            return "Fine. Be that way!";
        char punc = sTrim[sTrim.Length - 1];
        if (statement.ToUpper() == statement && !(statement.ToLower() == statement))
            return punc == '?' ? "Calm down, I know what I'm doing!" : "Whoa, chill out!";
        else if (punc == '?')
            return "Sure.";
        else
            return "Whatever.";

    }
}